presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -681,6 +683,11 @@ public ListeningExecutorService createResourceManagerExecutor(ResourceManagerCon
                 config -> NOOP.equalsIgnoreCase(config.getTracerType()),
                 moduleBinder -> moduleBinder.bind(TracerProvider.class).to(NoopTracerProvider.class).in(Scopes.SINGLETON)));
 
+        install(installModuleIf(
+                TracingConfig.class,
+                config -> TESTING.equalsIgnoreCase(config.getTracerType()),
+                moduleBinder -> moduleBinder.bind(TracerProvider.class).to(TestingTracerProvider.class).in(Scopes.SINGLETON)));
```

Comment:
By looking at your unit tests, do we need to install TestingTracerProvider in the server main module? The unit tests are mostly local by directly calling `new TestingTracerProvider()`. We don't launch a server for unit tests right? In the case, we can remove this code and put TestingTracerProvider to test folder.

Reply:
The main purpose for TestingTracerProvider is for debugging and testing in real server (or it can be put in integration tests). The unit test is to test the correctness of thie TestingTracerProvider class.